services/director/src/simcore_service_director/registry_proxy.py

```diff
@@ -155,68 +151,85 @@ async def _auth_registry_request(
         token_url = URL(auth_details["realm"]).with_query(
             service=auth_details["service"], scope=auth_details["scope"]
         )
-        async with session.get(token_url, auth=auth, **kwargs) as token_resp:
-            if token_resp.status != HTTPStatus.OK:
-                msg = f"Unknown error while authentifying with registry: {token_resp!s}"
-                raise RegistryConnectionError(msg=msg)
-            bearer_code = (await token_resp.json())["token"]
-            headers = {"Authorization": f"Bearer {bearer_code}"}
-            async with getattr(session, method.lower())(
-                url, headers=headers, **kwargs
-            ) as resp_wtoken:
-                if resp_wtoken.status == HTTPStatus.NOT_FOUND:
-                    logger.exception("path to registry not found: %s", url)
-                    raise ServiceNotAvailableError(service_name=f"{url}")
-                if resp_wtoken.status > 399:
-                    logger.exception(
-                        "Unknown error while accessing with token authorized registry: %s",
-                        str(resp_wtoken),
-                    )
-                    raise RegistryConnectionError(msg=f"{resp_wtoken}")
-                resp_data = await resp_wtoken.json(content_type=None)
-                resp_headers = resp_wtoken.headers
-                return (resp_data, resp_headers)
-    elif auth_type == "Basic":
+        token_resp = await session.get(f"{token_url}", auth=auth, **kwargs)
```

Comment:
Now that you installed httpx, you might want to use [Custom auth schemes](https://www.python-httpx.org/advanced/authentication/#custom-authentication-schemes) . Here is an example I did one that implements the bearer auth-flow

https://github.com/itisfoundation/osparc-simcore/blob/5732a12e07e63d5ce55010ede9b9ab543bb9b278/services/payments/scripts/example_payment_gateway.py#L141-L167

Reply:
yes this looks nice, but I cannot test it unless I have a portus system. therefore I am tempted to just remove it